spec/integration/worker_async_spec.rb

```diff
@@ -52,7 +52,7 @@ module Backend
         end
 
         worker = Worker.new(async: true, job_fetcher: JobFetcher.new(fetch_timeout: 1))
-        worker_thread = Thread.new { worker.work }
+        worker_thread = Async { worker.work }
```

Comment:
it turns out that in CI, using a thread can cause the deadlock you have observed locally, although I still couldn't reproduce on my machine.. just to tell you it is not only you

Reply:
Maybe this should be called `worker_fiber` now

Reply:
not actually a fiber, renamed to `worker_async_job`

Reply:
finally `worker_async_task`